packages/server-reason-react-ppx/test/test.re

```diff
@@ -428,9 +464,85 @@ let multiple_contexts = () => {
   );
 };
 
-Alcotest.run(
+module FunctionReferences: ReactServerDOM.FunctionReferences = {
+  type t = Hashtbl.t(string, ReactServerDOM.server_function);
+
+  let registry = Hashtbl.create(10);
+  let register = Hashtbl.add(registry);
+  let get = Hashtbl.find_opt(registry);
+};
+
+module ServerFunction = {
+  [@react.server.function]
+  let simpleResponse = (~name: string, ~age: int): Js.Promise.t(string) => {
+    Lwt.return(Printf.sprintf("Hello %s, you are %d years old", name, age));
+  };
+};
+
+let server_function_reference = () => {
+  let%lwt route_response =
+    FunctionReferences.get(ServerFunction.simpleResponse.id)
+    |> (
+      fun
+      | Some(Body(handler)) => handler([|`String("John"), `Int(30)|])
+      | _ => assert(false)
+    );
+
+  switch (route_response) {
+  | React.Json(json) =>
+    assert_string(string_of_json(json), "Hello John, you are 30 years old");
+    Lwt.return_unit;
+  | _ => failwith("Expected a JSON response")
+  };
+};
+
+let server_function_reference_args_error = () => {
+  (
+    try(
+      FunctionReferences.get(ServerFunction.simpleResponse.id)
+      |> (
+        fun
+        | Some(Body(handler)) => handler([|`String("John"), `Int(30)|])
+        | _ => assert(false)
+      )
+      |> ignore
+    ) {
+    | Failure(error) =>
+      assert_string(
+        error,
+        "server-reason-react: error on decoding argument 'age'. EXPECTED: int, RECEIVED: \"30\"",
+      )
+      |> ignore
+    }
+  )
+  |> ignore;
+  Lwt.return_unit;
+};
+
+let server_function = () => {
+  let%lwt response =
+    ServerFunction.simpleResponse.call(~name="John", ~age=30);
+  assert_string(response, "Hello John, you are 30 years old");
+  Lwt.return_unit;
+};
+
+let server_function_curried = () => {
+  let johnFuntion = ServerFunction.simpleResponse.call(~name="John");
+  let%lwt response = johnFuntion(~age=30);
+  assert_string(response, "Hello John, you are 30 years old");
+  Lwt.return_unit;
+};
+
+Alcotest_lwt.run(
   "server-reason-react.ppx",
   [
+    test_lwt("server_function_reference", server_function_reference),
+    test_lwt(
+      "server_function_reference_args_error",
+      server_function_reference_args_error,
+    ),
+    test_lwt("server_function", server_function),
+    test_lwt("server_function_curried", server_function_curried),
```

Comment:
What are we really testing here?

Reply:
Ops, this was one test that I added when we were talking about the formData and bind, and I forgot to clean it.